### PR TITLE
JAMES-2813 expose node informations in webadmin execution details

### PR DESCRIPTION
**Pull request by rouazana**

### CODE REVIEW

File: server/protocols/webadmin/webadmin-core/src/test/java/org/apache/james/webadmin/routes/TasksRoutesTest.java [hunk 1]
```diff
@@ -22,11 +22,7 @@
 import static io.restassured.RestAssured.given;
 import static io.restassured.RestAssured.when;
 import static io.restassured.RestAssured.with;
-import static org.hamcrest.Matchers.empty;
-import static org.hamcrest.Matchers.hasSize;
-import static org.hamcrest.Matchers.is;
-import static org.hamcrest.Matchers.isOneOf;
-import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.*;
```
Comment (blackheaven) [inline server/protocols/webadmin/webadmin-core/src/test/java/org/apache/james/webadmin/routes/TasksRoutesTest.java]: no wildcard

**Comment by blackheaven** [comment]
It is JAMES-28*7*3

File: server/protocols/webadmin/webadmin-core/src/test/java/org/apache/james/webadmin/routes/TasksRoutesTest.java (hunk 1 above)
Comment (rouazana) [reply server/protocols/webadmin/webadmin-core/src/test/java/org/apache/james/webadmin/routes/TasksRoutesTest.java]: sorry, I'm an Intellij newbie, I need to configure it.

**Comment by rouazana** [comment]
> It is JAMES-2873

oops, fixed.

**Comment by trantienduchn** [comment]
```
[83f6043722d10968b5c4808a5cbda0b3711956af] [ERROR] src/main/java/org/apache/james/webadmin/dto/ExecutionDetailsDto.java:[32] (imports) CustomImportOrder: Import statement for 'org.apache.james.task.eventsourcing.Hostname' is in the wrong order. Should be in the 'SPECIAL_IMPORTS' group, expecting not assigned imports on this line.
```

File: server/protocols/webadmin/webadmin-core/src/test/java/org/apache/james/webadmin/routes/TasksRoutesTest.java (hunk 1 above)
Comment (Arsnael) [reply server/protocols/webadmin/webadmin-core/src/test/java/org/apache/james/webadmin/routes/TasksRoutesTest.java]: Finally joined the dark side of IDEs? :)

**Comment by Arsnael** [comment]
You need to configure your imports ordering too :)

**Comment by rouazana** [comment]
```
[e1942aead6c4dd64aa7467be120ea4704c701502] [ERROR] Failures: 
[e1942aead6c4dd64aa7467be120ea4704c701502] [ERROR]   ICSAttachmentWorkflowTest.mailShouldNotContainCalendarContentInTextBodyButAttachment:801 
[e1942aead6c4dd64aa7467be120ea4704c701502] Expecting:
[e1942aead6c4dd64aa7467be120ea4704c701502]  <"AAAD OK FETCH completed.
[e1942aead6c4dd64aa7467be120ea4704c701502] ">
[e1942aead6c4dd64aa7467be120ea4704c701502] to contain:
[e1942aead6c4dd64aa7467be120ea4704c701502]  <["Content-Type: multipart/mixed", "Content-Disposition: attachment"]>
[e1942aead6c4dd64aa7467be120ea4704c701502] but could not find:
[e1942aead6c4dd64aa7467be120ea4704c701502]  <["Content-Type: multipart/mixed", "Content-Disposition: attachment"]>
[e1942aead6c4dd64aa7467be120ea4704c701502]  
[e1942aead6c4dd64aa7467be120ea4704c701502] [INFO] 
[e1942aead6c4dd64aa7467be120ea4704c701502] [ERROR] Tests run: 188, Failures: 1, Errors: 0, Skipped: 1
```
test this please

File: server/protocols/webadmin/webadmin-core/src/test/java/org/apache/james/webadmin/routes/TasksRoutesTest.java (hunk 1 above)
Comment (rouazana) [reply server/protocols/webadmin/webadmin-core/src/test/java/org/apache/james/webadmin/routes/TasksRoutesTest.java]: Eclipse is broken with current James code source :/ (it does not appreciate the mix of Java and Scala)

File: server/protocols/webadmin/webadmin-core/src/test/java/org/apache/james/webadmin/routes/TasksRoutesTest.java [hunk 2]
```diff
@@ -99,7 +101,10 @@ void listShouldReturnTaskDetailsWhenTaskInProgress() throws Exception {
             .body("", hasSize(1))
             .body("[0].status", is(TaskManager.Status.IN_PROGRESS.getValue()))
             .body("[0].taskId", is(taskId.asString()))
-            .body("[0].class", is(not(empty())));
+            .body("[0].class", is(not(empty())))
+            .body("[0].submittedNode", is(HOSTNAME))
+            .body("[0].ranNode", is(HOSTNAME))
```
Comment (mbaechler) [inline server/protocols/webadmin/webadmin-core/src/test/java/org/apache/james/webadmin/routes/TasksRoutesTest.java]: executedOn?

File: server/protocols/webadmin/webadmin-core/src/test/java/org/apache/james/webadmin/routes/TasksRoutesTest.java [hunk 3]
```diff
@@ -99,7 +101,10 @@ void listShouldReturnTaskDetailsWhenTaskInProgress() throws Exception {
             .body("", hasSize(1))
             .body("[0].status", is(TaskManager.Status.IN_PROGRESS.getValue()))
             .body("[0].taskId", is(taskId.asString()))
-            .body("[0].class", is(not(empty())));
+            .body("[0].class", is(not(empty())))
+            .body("[0].submittedNode", is(HOSTNAME))
```
Comment (mbaechler) [inline server/protocols/webadmin/webadmin-core/src/test/java/org/apache/james/webadmin/routes/TasksRoutesTest.java]: submittedFrom?

File: server/protocols/webadmin/webadmin-core/src/test/java/org/apache/james/webadmin/routes/TasksRoutesTest.java [hunk 4]
```diff
@@ -99,7 +101,10 @@ void listShouldReturnTaskDetailsWhenTaskInProgress() throws Exception {
...
+            .body("[0].ranNode", is(HOSTNAME))
+            .body("[0].cancelRequestedNode", nullValue());
```
Comment (mbaechler) [inline server/protocols/webadmin/webadmin-core/src/test/java/org/apache/james/webadmin/routes/TasksRoutesTest.java]: cancelledFrom?

**Comment by blackheaven** [comment]
test this please

**Comment by blackheaven** [comment]
test this please

**Comment by chibenwa** [comment]
```

[010d55fcc7f995d44552c6ab6902d3797cf55fc6] Step 4/22 : ADD destination/james-server-app-3.4.0-SNAPSHOT-app.zip /root/james-server-app-3.4.0-SNAPSHOT-app.zip
[010d55fcc7f995d44552c6ab6902d3797cf55fc6] ADD failed: stat /var/lib/docker/tmp/docker-builder469788429/destination/james-server-app-3.4.0-SNAPSHOT-app.zip: no such file or directory
```

Rebase needed

**Comment by rouazana** [comment]
> "You have triggered an abuse detection mechanism. Please wait a few minutes before you try again."

test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**